runtime/queries/column_timeseries.go

```diff
@@ -385,11 +397,21 @@ func getFilterFromDimensionValuesFilter(
 		if notNulls {
 			inClause := escapedName + " " + prefix + " IN ("
 			for j, iv := range dv.In {
-				if _, ok := iv.Kind.(*structpb.Value_NullValue); !ok {
-					inClause += "'" + EscapeSingleQuotes(iv.GetStringValue()) + "'"
-					if j < len(dv.In)-1 {
-						inClause += ", "
-					}
+				switch iv.Kind.(type) {
+				case *structpb.Value_StringValue:
+
+					inClause += "'" + escapeSingleQuotes(iv.GetStringValue()) + "'"
+				case *structpb.Value_NumberValue:
+					inClause += "?"
+					args = append(args, iv.GetNumberValue())
+				case *structpb.Value_NullValue:
+					continue
+				case *structpb.Value_BoolValue:
+					inClause += "?"
+					args = append(args, iv.GetBoolValue())
+				}
```

Comment:
Remove the null case (should be ensured by previous `if notNulls`), and add a default case that `panic`s if reached (since all cases should be covered)

Reply:
no, `notNulls` is only and indicator of non-nulls values, this block has to filter them out

Reply:
I see, makes sense!